### PR TITLE
Modify getup to use raw sensors average

### PR DESCRIPTION
**Pull request by tayloryoung6396**
This PR is dependant on #464 

Works on the robot

### CODE REVIEW

File: module/behaviour/skills/Getup/src/Getup.cpp [hunk 1]
```diff
@@ -65,30 +66,48 @@ namespace module::behaviour::skills {
...
+                }
+                acc_reading.normalize();
+
+                // check that the accelerometer reading is less than some predetermined
+                // amount
+                if (!gettingUp && std::acos(EigenVector3d::UnitZ().dot(acc_reading)) > FALLEN_ANGLE) {
+                    isFront = false;
+                    if(pi/2 - std::acos(EigenVector3d::UnitX().dot(acc_reading)) > 0.0){
+                        isFront = true;
+                    }
```
Comment (Bidski) [inline module/behaviour/skills/Getup/src/Getup.cpp:86]:
```suggestion
                    // If we are on our side (== 0.0) then we want to execute the rollover
                    isFront = (M_PI_2 - std::acos(EigenVector3d::UnitX().dot(acc_reading)) >= 0.0);
```

File: module/behaviour/skills/Getup/src/Getup.cpp [hunk 2]
```diff
@@ -65,30 +66,48 @@ namespace module::behaviour::skills {
...
+
+                    updatePriority(GETUP_PRIORITY);
+                    fallenCheck.disable();
+                }
+            });
+
+        on<Trigger<ExecuteGetup>, With<RawSensors>>().then(
+            "Execute Getup",
+            [this](const std::list<std::shared_ptr<const RawSensors>>& sensors) {
```
Comment (Bidski) [inline module/behaviour/skills/Getup/src/Getup.cpp:95]:
```suggestion
        on<Trigger<ExecuteGetup>>().then(
            "Execute Getup",
            [this]() {
```

**Review by Bidski — CHANGES_REQUESTED** [review]
Format things too

File: module/behaviour/skills/Getup/src/Getup.cpp (hunk 2 above)
Comment (tayloryoung6396) [reply module/behaviour/skills/Getup/src/Getup.cpp:95]: I don't have clang.. :P

File: module/behaviour/skills/Getup/src/Getup.cpp [hunk 3]
```diff
@@ -65,19 +65,34 @@ namespace module::behaviour::skills {
...
-        });
+        fallenCheck = on<Last<20, Trigger<RawSensors>>, Single>().then(
+            "Getup Fallen Check",
+            [this](const std::list<std::shared_ptr<const RawSensors>>& sensors) {
+                Eigen::Vector3d acc_reading = Eigen::Vector3d::Zero();
+
+                for (const auto& s : sensors) {
+                    acc_reading += Eigen::Vector3d(s->accelerometer.x, s->accelerometer.y, s->accelerometer.z);
+                }
+                acc_reading.normalize();
```
Comment (Bidski) [inline module/behaviour/skills/Getup/src/Getup.cpp:76]: The more I think about it, the more I feel like we actually need to do this
```suggestion
                acc_reading = (acc_reading / double(sensors.size())).normalized();
```

File: module/behaviour/skills/Getup/src/Getup.cpp [hunk 4]
```diff
@@ -65,19 +65,34 @@ namespace module::behaviour::skills {
...
+
+                // check that the accelerometer reading is less than some predetermined
+                // amount
+                if (!gettingUp && std::acos(EigenVector3d::UnitZ().dot(acc_reading)) > FALLEN_ANGLE) {
+                    isFront = false;
+                    if (pi / 2 - std::acos(EigenVector3d::UnitX().dot(acc_reading)) > 0.0) {
+                        isFront = true;
+                    }
```
Comment (Bidski) [inline module/behaviour/skills/Getup/src/Getup.cpp:84]:
```suggestion
                    // If we are on our side, treat it as being on our front, hopefully the rollover will help matters
                    isFront = (M_PI_2 - std::acos(EigenVector3d::UnitX().dot(acc_reading)) >= 0.0);
```

**Review by Bidski — CHANGES_REQUESTED** [review]

**Review by Bidski — DISMISSED** [review]
Looks ok

**Comment by tayloryoung6396** [comment]
Works on the robot, someone should test it in webots

**Review by Bidski — COMMENTED** [review]
LGTM. Pending sim testing

**Comment by ysims** [comment]
Does appear to work in webots.

**Comment by tayloryoung6396** [comment]
> Does appear to work in webots.

You could have approved it..

**Review by ysims — APPROVED** [review]
Seems good enough.

**Pull request merged**